addons/payment_stripe/controllers/main.py

```diff
@@ -19,9 +18,45 @@
 
 class StripeController(http.Controller):
     _checkout_return_url = '/payment/stripe/checkout_return'
+    _onboarding_return_url = '/payment/stripe/onboarding/return/'
+    _onboarding_refresh_url = '/payment/stripe/onboarding/refresh/'
     _validation_return_url = '/payment/stripe/validation_return'
     WEBHOOK_AGE_TOLERANCE = 10*60  # seconds
 
+    @http.route(
+        _onboarding_return_url + '<int:acquirer_id>/<int:menu_id>', type='http', auth='public'
+    )
+    def stripe_onboarding_return(self, acquirer_id, menu_id):
+        """Stripe return URL
+
+            This route is used by Stripe to return after or during the onboarding.
+            Once called, the controller redirects to the payment acquirer.
+
+            :returns: Redirection to Stripe Acquirer Form
+        """
+        action = request.env.ref(
+            'payment_stripe.action_payment_acquirer_onboarding', raise_if_not_found=False
+        )
+        return request.redirect('/web?#action=%s&id=%s&menu_id=%s' % (
+            action and action.id or 'payment.action_payment_acquirer', acquirer_id, menu_id
+        ))
```

Comment:
f-string power!
```suggestion
       return request.redirect(f'/web?#action={action.id}&id={acquirer_id}&menu_id={menu_id}')
```
Note that following the addition of the fallback for the menu, I removed the fallback for the action.

Reply:
Why do we remove the fallback for the action here ? Since the stripe connect onboarding begins directly from the onboarding panel and we changed the behavior of the action in the python, user could be redirected here without having the action stored so we need to fallback on the `payment.action_payment_acquirer` action.

Reply:
My comment made sense two days ago because the menu would have a fallback but it's good how you changed it now.